server/src/main/java/org/elasticsearch/cluster/metadata/Template.java

```diff
@@ -132,8 +149,13 @@ public Template(StreamInput in) throws IOException {
         } else {
             this.aliases = null;
         }
-        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0) && DataLifecycle.isEnabled()) {
-            this.lifecycle = in.readOptionalWriteable(DataLifecycle::new);
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_006) && DataLifecycle.isEnabled()) {
+            boolean isExplicitNull = in.readBoolean();
+            if (isExplicitNull) {
+                this.lifecycle = NO_LIFECYCLE;
+            } else {
+                this.lifecycle = in.readOptionalWriteable(DataLifecycle::new);
+            }
         } else {
             this.lifecycle = null;
         }
```

Comment:
I think we need to make this change in a BWC way to not break the BWC tests with `8_8_0` ? 

i.e. keep `if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_8_0)` as well

Reply:
Yes, I added that